lib/action_view/component/railtie.rb

```diff
@@ -42,6 +42,12 @@ class Railtie < Rails::Railtie # :nodoc:
         end
       end
 
+      initializer "action_view_component.compile_templates" do
+        ActiveSupport.on_load(:after_initialize) do
+          ActionView::Component::Base.descendants.each(&:compile) if config.eager_load
```

Comment:
I'm not convinced it's strictly necessary to check for eager loading here since `.descendants` would most likely return an empty array if the classes are lazily loaded. I included it for good measure, as I find it a bit more intention revealing.